core/src/main/java/org/sonarsource/sonarlint/core/issue/IssueServiceImpl.java

```diff
@@ -175,28 +178,49 @@ public CompletableFuture<CheckStatusChangePermittedResponse> checkStatusChangePe
+    isSonarCloudOrSQ104orNewer = isSonarCloudOrSQ104orNewer || is104orNewer(connectionId, serverApi);
+
+    // we need a final or effectively final variable for the lambda
+    var useNewStatuses = isSonarCloudOrSQ104orNewer;
+
     return asUUID(issueKey)
       .flatMap(localOnlyIssueRepository::findByKey)
       .map(r -> {
         var anticipateTransitionsSupported = checkAnticipatedStatusChangeSupported(serverApi, connectionId);
         // no way to easily check if 'Administer Issue' permission is granted, might fail later
-        return CompletableFuture.completedFuture(toResponse(anticipateTransitionsSupported, UNSUPPORTED_SQ_VERSION_REASON));
+        return CompletableFuture.completedFuture(toResponse(anticipateTransitionsSupported, UNSUPPORTED_SQ_VERSION_REASON, useNewStatuses));
       })
       .orElseGet(() -> serverApi.issue().searchByKey(params.getIssueKey())
-        .thenApply(issue -> toResponse(hasAdministerIssuePermission(issue), STATUS_CHANGE_PERMISSION_MISSING_REASON)));
+        .thenApply(issue -> toResponse(hasAdministerIssuePermission(issue, useNewStatuses), STATUS_CHANGE_PERMISSION_MISSING_REASON, useNewStatuses)));
   }
 
-  private static CheckStatusChangePermittedResponse toResponse(boolean permitted, String reason) {
+  private boolean is104orNewer(String connectionId, ServerApi serverApi) {
+    var serverVersionSynchronizer = new ServerInfoSynchronizer(storageService.connection(connectionId));
+    var serverVersion = serverVersionSynchronizer.readOrSynchronizeServerInfo(serverApi);
+    return serverVersion.getVersion().compareToIgnoreQualifier(SQ_ACCEPTED_TRANSITION_MIN_VERSION) >= 0;
+  }
+
+  private static CheckStatusChangePermittedResponse toResponse(boolean permitted, String reason, boolean useNewStatuses) {
     return new CheckStatusChangePermittedResponse(permitted,
       permitted ? null : reason,
       // even if not permitted, return the possible statuses, if clients still want to show users what's supported
-      Arrays.asList(ResolutionStatus.values()));
+      Arrays.asList(useNewStatuses ? ResolutionStatus.get104orNewerStatuses() : ResolutionStatus.get103orOlderStatuses()));
   }
 
-  private static boolean hasAdministerIssuePermission(Issues.Issue issue) {
+  private static boolean hasAdministerIssuePermission(Issues.Issue issue, boolean useNewStatuses) {
     // the 2 required transitions are not available when the 'Administer Issues' permission is missing
     // normally the 'Browse' permission is also required, but we assume it's present as the client knows the issue key
     var possibleTransitions = new HashSet<>(issue.getTransitions().getTransitionsList());
+
+    final Set<String> requiredTransitions;
+    if (useNewStatuses) {
+      requiredTransitions = Set.of(Transition.ACCEPT.getStatus(), Transition.FALSE_POSITIVE.getStatus());
+    } else {
+      requiredTransitions = Set.of(Transition.WONT_FIX.getStatus(), Transition.FALSE_POSITIVE.getStatus());
+    }
```

Comment:
I have the feeling that we have duplicated knowledge here and in the enum: there are 2 places where we configure the list of allowed statuses. Maybe we could have the mapping only defined once?